### PR TITLE
Raincatch 1174

### PR DESCRIPTION
**Pull request by jstaffor**
@austincunningham  / @witmicko

1) Ignore changes "docs/workforce-management-framework/topics/ref-passportauth-securitystrategy.adoc" - just house keeping changes required due to keycloak work.

If you want to have a call to go through the changes - no porbs at all, just let me know.

### CODE REVIEW

File: docs/workforce-management-framework/topics/pro-keycloak-enablement.adoc [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+----
+    // Comment to disable passport auth
+    require('@raincatcher/demo-auth-passport')('app'),
+    // Uncomment to enable Keycloak
+    // require('./keycloak');
+----
++
+. At this point, the _Routes_ are protected by *{WFM-RC-KeycloakName}*.
+
+<ENG: NOT SURE ABOUT THE NEXT TEXT APPEARING HERE>
```
Comment (witmicko) [inline docs/workforce-management-framework/topics/pro-keycloak-enablement.adoc:96]: @austincunningham @wtrocki

File: docs/workforce-management-framework/topics/pro-keycloak-implementation.adoc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+Import *{WFM-RC-KeycloakName}* _init_ <ENG: how?> from link:{WFM-RC-CoreURL}{WFM-RC-Branch}/demo/server/src/modules/keycloak/index.ts[Keycloak].
+
+[id='pro-applying-keycloak-init-{chapter}']
+[discrete]
+=== Applying {WFM-RC-KeycloakName} _init_
+
+Apply *{WFM-RC-KeycloakName}* _init_ to the security Middleware <ENG: what is 'security Middleware' and how?> to enable *{WFM-RC-KeycloakName}* for route protection.
+See the link:{WFM-RC-CoreURL}{WFM-RC-Branch}/demo/server/src/modules/index.ts[securitySetup] code for more information.
+
```
Comment (witmicko) [inline docs/workforce-management-framework/topics/pro-keycloak-implementation.adoc]: all info is provided in the linkes `securitySetup` file
middleware https://github.com/feedhenry-raincatcher/raincatcher-core/blob/983cd8c607cd417975acca86b1eb7d71dc40b039/demo/server/src/modules/index.ts#L35

how to enable : https://github.com/feedhenry-raincatcher/raincatcher-core/blob/983cd8c607cd417975acca86b1eb7d71dc40b039/demo/server/src/modules/index.ts#L55
These concepts should be clear for people with express.js basic experience (>99% of node devs)

File: docs/workforce-management-framework/topics/pro-keycloak-implementation.adoc [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+=== Importing The EndpointSecurity Interface
+
+Import the link:{WFM-RC-CoreURL}{WFM-RC-Branch}/cloud/auth/src/EndpointSecurity.ts[EndpointSecurity Interface] and apply it to a variable <ENG: what variable and how?>.
+
+[id='pro-importing-keycloak-init-{chapter}']
+[discrete]
+=== Importing {WFM-RC-KeycloakName} _init_
+
+Import *{WFM-RC-KeycloakName}* _init_ <ENG: how?> from link:{WFM-RC-CoreURL}{WFM-RC-Branch}/demo/server/src/modules/keycloak/index.ts[Keycloak].
+
```
Comment (witmicko) [inline docs/workforce-management-framework/topics/pro-keycloak-implementation.adoc]: just a code statement e.g. [`import { init as initKeycloak } from './keycloak';`](https://github.com/feedhenry-raincatcher/raincatcher-core/blob/983cd8c607cd417975acca86b1eb7d71dc40b039/demo/server/src/modules/index.ts#L14)

File: docs/workforce-management-framework/topics/pro-keycloak-implementation.adoc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+=== Initializing {WFM-RC-KeycloakName}
+
+See the link:{WFM-RC-CoreURL}{WFM-RC-Branch}/demo/server/src/modules/keycloak/index.ts[Keycloak code] for reference.
+
+[id='pro-importing-the-endpointsecurity-interface-{chapter}']
+[discrete]
+=== Importing The EndpointSecurity Interface
+
+Import the link:{WFM-RC-CoreURL}{WFM-RC-Branch}/cloud/auth/src/EndpointSecurity.ts[EndpointSecurity Interface] and apply it to a variable <ENG: what variable and how?>.
+
```
Comment (witmicko) [inline docs/workforce-management-framework/topics/pro-keycloak-implementation.adoc:30]: I think this is an example https://github.com/feedhenry-raincatcher/raincatcher-core/blob/983cd8c607cd417975acca86b1eb7d71dc40b039/demo/server/src/modules/keycloak/index.ts#L1
@wtrocki can you please confirm this?

File: docs/workforce-management-framework/topics/ref-keycloak-securitystrategy.adoc [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+To see an example of how roles are protected, see the *{WFM-RC-KeycloakName}* link:./pro-Keycloak-implementaion.adoc[documentation].
+
+[id='defining-keycloak-password-storage-{chapter}']
+== Defining {WFM-RC-KeycloakName} Password Storage
+*{WFM-RC-KeycloakName}* uses the link:https://en.wikipedia.org/wiki/PBKDF2[BKDF2 algorithm] for password hashing by default.
+*{WFM-RC-KeycloakName}* also provides enforced link:{WFM-RC-KeycloakURL}server_admin/topics/authentication/password-policies.html[password policies] from the server.
+
+<ENG:Can you expand just a little on this? "from the server? - what does this mean?", "enforced passporot policies - does this mean one has to login using a password and username by default when using the Client App">
+
```
Comment (witmicko) [inline docs/workforce-management-framework/topics/ref-keycloak-securitystrategy.adoc:31]: http://www.keycloak.org/docs/2.1/server_admin_guide/topics/authentication/password-policies.html
Password policy is set of rules that all passwords must comply with - min length, special characters etc.

**Comment by jstaffor** [comment]
@austincunningham , I'm blocked by this also so can you take a look at this after your lunch today? This is an example where I may have changed to much and I am happy to take a call on this and work together to ensure we are both happy with what is here - even if it means we end up reverting certain content.

**Review by austincunningham — APPROVED** [review]
approve

**Comment by witmicko** [comment]
@jstaffor I think we need to revert this and edit out those `<ENG:` comments

**Comment by wtrocki** [comment]
@witmicko @austincunningham - Quickly checked documentation. 
Let's leave that for now as @austincunningham can really reject that from his branch.
We may focus on bringing things to master and then myself, Paolo and Jameel will review them and see where we missing things. Documentation for keycloak may need to focus on some different aspects than we have here, but PR's it's really restricted to suggest this.

**Pull request merged**